### PR TITLE
Initial version of peer discovery docs

### PR DESCRIPTION
**Pull request by dshulyak**

### CODE REVIEW

**Comment by dshulyak** [comment]
idk yet if it is best to use wiki or not, also i want to improve pictures a bit

@themue @adambabik let me know what else you would like to see in the docs

File: _assets/docs/discovery.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Peer will run search loop quite often (every 500ms at the time of writing) with the
+goal to visit as much nodes as possible and once it will find minimum amount of
+peers with required capability it will swich to slow mode.
+2. Slow mode.
+This mode might be useful to get information about new peers that can be used
+later if initial set of peers will dissapear. Should run once in a 10m-30m.
+
+However finding peers even with fast mode can take noticeable amount of time,
+which is fine if node is a long-running, but will be a huge problem on a mobile
+device. To workaround such problem we will introduce a leveldb cache that will
```
Comment (themue) [inline _assets/docs/discovery.md:47]: 👍

**Review by themue — APPROVED** [review]
Very good description.

**Review by divan — APPROVED** [review]

File: _assets/docs/discovery.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+Peer pool in status-go
+----------------------
+
+Peer pool uses search mechanism described above to find peers in a timely manner
+without using a lot of resources in the long run. It achieves it by introducing
+two modes of synchronization:
+
+1. Fast mode.
+Peer will run search loop quite often (every 500ms at the time of writing) with the
```
Comment (mandrigin) [inline _assets/docs/discovery.md:38]: We can add a link to the source file where that constant is defined.

**Review by mandrigin — APPROVED** [review]
🏅

**Review by adambabik — APPROVED** [review]
For now, I think it's perfect 👏

**Comment by mandrigin** [comment]
@dshulyak is that one still in progress? or can we merge this iteration for now?

**Comment by dshulyak** [comment]
merged, will improve images later

**Pull request merged**